openmdao/utils/general_utils.py

```diff
@@ -994,6 +997,29 @@ def _is_slice(indices):
     return any(isinstance(i, slice) for i in indices)
 
 
+def _is_ellipsis(indices):
```

Comment:
I think you should probably combine `_is_ellipsis` and `is_slice` into one function.  They always seem to be used together, so it will save code.

Note, there are some other places in the model where `is_slice` is used, but you haven't added `is_ellipsis`, so you should check these carefully.  In particular, `default_transfer.py` and `petsc_transfer.py`.